externals/llvm-external-projects/torch-mlir-dialects/include/torch-mlir-dialects/Conversion/TcpToTosa/TcpToTosa.h

```diff
@@ -0,0 +1,20 @@
+#ifndef TORCH_MLIR_DIALECTS_CONVERSION_TCPTOTOSA_TCPTOTOSA_H_
+#define TORCH_MLIR_DIALECTS_CONVERSION_TCPTOTOSA_TCPTOTOSA_H_
+
+#include "mlir/Pass/Pass.h"
+
+namespace mlir {
+
+#define GEN_PASS_DECL_CONVERTTCPTOTOSA
+#include "torch-mlir-dialects/Conversion/Passes.h.inc"
+
+namespace torch {
+namespace tcp {
+
+std::unique_ptr<Pass> createConvertTcpToTosaPass();
```

Comment:
This is auto-generated right?  I'm looking here: https://github.com/llvm/llvm-project/blob/e854c17b02f8cd82a303d223ba5f3b0d87579cd7/mlir/tools/mlir-tblgen/PassGen.cpp#L127

Reply:
I tried removing this and it doesn't auto generate for me. Not sure if I'm missing something in cmake files to do this.

Reply:
Can you check if other passes have this?  Maybe you need to do something extra to put it in the `torch::tcp` namespace?

Reply:
I think what you pointed to is only generating code that goes into `Passes.h.inc` file. For example, that file contains the following:
```
inline void registerConvertTcpToTosa() {
  ::mlir::registerPass([]() -> std::unique_ptr<::mlir::Pass> {
    return mlir::torch::tcp::createConvertTcpToTosaPass();
  });
}
```
I think that is what is being generated here.

Reply:
@silvasean Do you know if this file could be generated automatically?

Reply:
createConvertTcpToTosaPass needs to be declared here. It is not autogenerated. See e.g. https://github.com/llvm/torch-mlir/blob/main/include/torch-mlir/Conversion/TorchToArith/TorchToArith.h